lib/src/model/trait.dart

```diff
@@ -43,29 +71,61 @@ class TraitWithIdentity {
   @JsonKey(name: 'trait_key')
   final String key;
   @JsonKey(
-      name: 'trait_value',
-      fromJson: nonNullStringFromJson,
-      toJson: stringToJson)
-  final String value;
+    name: 'trait_value',
+    fromJson: _fromJson,
+    toJson: _toJson,
+  )
+  final dynamic value;
 
   TraitWithIdentity({
     required this.identity,
     required this.key,
     required this.value,
   });
+
   factory TraitWithIdentity.fromJson(Map<String, dynamic> json) =>
       _$TraitWithIdentityFromJson(json);
 
   Map<String, dynamic> toJson() => _$TraitWithIdentityToJson(this);
+
   String asString() => jsonEncode(toJson());
+
   TraitWithIdentity copyWith({
     Identity? identity,
     String? key,
-    String? value,
+    dynamic value,
   }) =>
       TraitWithIdentity(
         identity: identity ?? this.identity,
         key: key ?? this.key,
         value: value ?? this.value,
       );
+
+  static dynamic _fromJson(dynamic jsonValue) {
+    if (jsonValue is num) {
+      if (jsonValue % 1 == 0) {
+        return jsonValue.toInt(); // Treat as int if it's a whole number
+      } else {
+        return jsonValue.toDouble(); // Treat as double if it's a decimal
+      }
+    } else if (jsonValue is String) {
+      return jsonValue;
+    } else if (jsonValue is bool) {
+      return jsonValue;
+    }
+    throw ArgumentError('Invalid value type');
+  }
+
+  static dynamic _toJson(dynamic value) {
```

Comment:
Seems like this function is just doing the same for all types? Can we not just check if it's not a supported type and raise an exception if so?

Reply:
Yeah, done